DLPJTS-97 New example draws boxes around each character.

datalogics-kam opened the pull request:
**Please do [pdf-java-toolkit#978](https://octocat.dlogics.com/datalogics/pdf-java-toolkit/pull/978) first (this PR may not build correctly until then either).**
#### Changes in this pull request
- Demonstrates the easier-to-use method of getting character quads in PDFJT 4.2.
- Requires PDFJT 4.2 and Talkeetna 6.
- This sample and its test are excluded from building with older versions.
#### Fulfills  [DLPJTS-97](https://jira.datalogics.com/browse/DLPJTS-97)
#### Checklist for approving this pull request

(PR creator amend this with more conditions if necessary)
- [x] There are **unit tests** for new/changed code, or a good explanation why this was not possible.
- [x] All **CI builders** have indicated success (Give them a few minutes to notice the pull request.)
- [x] The **Pull request title** has the JIRA issue numbers separated by spaces (if any), a space, and then a short, but descriptive summary.
- [x] **Commit messages** are well formed: [A note about Git commit messages](http://www.tpope.net/node/106)
- [x] New public packages, classes, and methods are **documented**. (Strongly consider documenting private classes and methods.)
#### Blockers

Add a checkbox for yourself with your **&#64;name** to this list if you are holding this PR open for review. PR Shepherd, please hold off merging this PR until these are all checked:
- [x] _&lt;add your name here with an **@**>_

datalogics-kam committed:
TextCharacterBoxes: New sample that draws a box around every character

It's a demonstration of using the improved character quad APIs in PDFJT
4.2.

- This sample requires Talkeetna 6 and PDFJT 4.2.
- Takes an input document, puts a thin red box around every character,
  and writes an output document.

datalogics-kam committed:
POM: Exclude TextCharacterBoxes before Talkeetna 6.

datalogics-kam committed:
SampleTest: Convert LF to CR in page contents

- Mirrors conversion to CR when reading the contents files from test
  resources.
- Some test documents were using LF as a line ending character in their
  contents.

datalogics-kam committed:
TextCharacterBoxesTest: Test the TextCharacterBoxes sample

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build failed.

datalogics-coreyy commented on pom.xml:
```diff
@@ -151,12 +151,18 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
-        <version>2.10.1</version>
+        <version>2.10.4</version>
```
Can you explain this bump here from 1 to 4?

datalogics-kam replied:
A lot of times when I have trouble getting things to work, I update to the latest plugin along the way, just to make sure.

datalogics-coreyy commented:
I'm not sure what the purpose of TextCharacterBoxesTest.pdf.page0.txt and TextCharacterBoxesTest.pdf.page1.txt serves.

datalogics-coreyy commented:
All looks good here and tests fine on my machine. I'd just ask @datalogics-robl to spot check the change in pom.xml. Are we concerned about the default check failing (probably just because the other pr wasn't merged yet)?

datalogics-kam committed:
POM: Add explanation for combine.self="override"

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build failed.

datalogics-kam committed:
RunMainWithArgsIntegrationTest - Don't run tests that don't have args

- The test uses reflection to find main methods, but some might
  not have an arg list set up.
- Can't refer to some tests that are only valid in some versions of
  PDFJT.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-coreyy commented:
I'll retest shortly here and if all goes well, well you know the rest
